src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -352,29 +352,26 @@ private static bool IsDictionaryTrainingPhase(CurrentIndexingScope currentIndexi
         {
             return currentIndexingScope != null && currentIndexingScope.Index.IsOnBeforeExecuteIndexing;
         }
-        
-        public object CreateVector(string fieldName, object value)
-        {
-            if (value is null or DynamicNullObject)
-                return null;
 
+        internal static IndexField RetrieveVectorField(string fieldName, object value)
+        {
             var currentIndexingScope = CurrentIndexingScope.Current;
-            if (IsDictionaryTrainingPhase(currentIndexingScope))
-                return null;
-            
-            // We're supporting two defaults:
-            // when Options are not set, we'll decide what is configuration in following manner:
-            // - value is textual or array of textual we're treating them as text input
-            // - otherwise, we will write as array of numerical values
             var fieldExists = currentIndexingScope.Index.Definition.IndexFields.TryGetValue(fieldName, out var indexField);
             if (fieldExists == false || indexField?.Vector is null)
             {
-                var isText = value is LazyStringValue or LazyCompressedStringValue or string or DynamicNullObject;
+                // We're supporting two defaults:
+                // when Options are not set, we'll decide what is configuration in following manner:
+                // - value is textual or array of textual we're treating them as text input
+                // - otherwise, we will write as array of numerical values
+                var isText = IsText(value);
                 if (isText == false && value is IEnumerable enumerable)
                 {
                     foreach (var item in enumerable)
                     {
-                        isText = item is LazyStringValue or LazyCompressedStringValue or string or DynamicNullObject;
+                        if (item is null or DynamicNullObject)
+                            continue;
+                        
+                        isText = item is LazyStringValue or LazyCompressedStringValue or string or DynamicNullObject or JsString;
```

Comment:
shouldn't you use here `IsText(item)`?

Reply:
This issue has been fixed in https://github.com/ravendb/ravendb/pull/19976 (which also contains this PR). Maybe we can supersede this PR with that one?